conda/gateways/disk/read.py

```diff
@@ -49,7 +49,7 @@ def yield_lines(path):
                 yield line
     except (IOError, OSError) as e:
         if e.errno == ENOENT:
-            raise StopIteration
+            pass
```

Comment:
`_vendor.toolz` also pukes out a `StopIteration` -- this should be fixed by replacing it by `toolz 0.8.1` or above